### PR TITLE
Update realm-core to v12.11.0

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

**Review by desistefanova — APPROVED** [review]

**Review by blagoev — COMMENTED** [review]
Should we add some changelog entries here for the core fixes
>Fix a use-after-free if the last external reference to an encrypted Realm was closed between when a client reset error was received and when the download of the new Realm began.
this one seems deserving an entry

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,13 +6,15 @@
 * Added `MutableSubscriptionSet.removeByType` for removing subscriptions by their realm object type. (Issue [#317](https://github.com/realm/realm-dart/issues/317))
 
 ### Fixed
-* None
+* CompensatingWriteErrorInfo reported string primary keys as boolean values instead. (Core upgrade)
```
Comment (nirinchev) [inline CHANGELOG.md:9]: This is not exposed in the dart SDK.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,13 +6,15 @@
...
+* Fix a use-after-free if the last external reference to an encrypted Realm was closed between when a client reset error was received and when the download of the new Realm began. (Core upgrade)
+* Fixed an assertion failure during client reset with recovery when recovering a list operation on an embedded object that has a link column in the path prefix to the list from the top level object. (Core upgrade)
```
Comment (nirinchev) [inline CHANGELOG.md:11]: This is not exposed either, so users wouldn't have experienced it.

Comment (nielsenko) [reply CHANGELOG.md]: It now says:
> ### Fixed
>* Fix a use-after-free if the last external reference to an encrypted Realm was closed between when a client reset error was received and when the download of the new Realm began. (Core upgrade)

Comment (nirinchev) [reply CHANGELOG.md]: The thing is, we don't have client reset with recovery, so there's no way for users to have gotten in this situation. This is only for client reset with recovery and doesn't affect the manual client reset mechanism.

**Review by nirinchev — APPROVED** [review]

**Review by desistefanova — APPROVED** [review]

**Review by blagoev — APPROVED** [review]

**Pull request merged**